src/Microsoft.DotNet.Darc/src/DarcLib/RemoteRepoBase.cs

```diff
@@ -298,6 +288,60 @@ private static void CheckoutSubmodules(LibGit2Sharp.Repository repo, LibGit2Shar
             }
         }
 
+        private static void SafeCheckout(LibGit2Sharp.Repository repo, string commit, LibGit2Sharp.CheckoutOptions options, ILogger log)
```

Comment:
How similar is this to the other SafeCheckout? Can we put the code in a common place if it's similar enough?

Reply:
Consolidated this in a helper class.